service-chain-aws/deploy-multi_layer-servicechain/common.tf

```diff
@@ -0,0 +1,169 @@
+data "aws_ami" "node_ami" {
+  most_recent = true
+  owners      = ["aws-marketplace"]
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+
+  filter {
+    name   = "image-type"
+    values = ["machine"]
+  }
+
+  filter {
+    name   = "name"
+    values = ["CentOS Linux 7*"]
+  }
+}
+
+resource "aws_security_group_rule" "ingress_en_l2_anchoring_sg_tcp" {
```

Comment:
I have some error like this. Can you add an example about en_security_group_id? 

`│ Error: InvalidGroupId.Malformed: Invalid id: "henry-sc-deploy-multi2-en" (expecting "sg-...")
│       status code: 400, request id: 29f8b94e-bcbf-408d-8939-4c709a4ac198
│
│   with aws_security_group_rule.ingress_en_l2_anchoring_sg_tcp,
│   on common.tf line 26, in resource "aws_security_group_rule" "ingress_en_l2_anchoring_sg_tcp":
│   26: resource "aws_security_group_rule" "ingress_en_l2_anchoring_sg_tcp" {
│
`

Reply:
You have to input the existing EN's security group ID. The security group ID is not something you can decide, but instead it is chosen by the AWS. Like the error message says, security group ID looks like `sg-0123456789abcdef`.

Suppose you have deployed L1+L2. If you want to deploy L3, you will use this terraform with `en_security_group_id` set to L2 EN's security group. The reason for requiring the parent chain EN's security group is to allow inbound connection for the bridge P2P port.

Reply:
Thanks you. I will try again and success.